contracts/src/Teleporter/upgrades/README.md

```diff
@@ -0,0 +1,66 @@
+# Upgradeability
+
+## Overview
+
+The `TeleporterMessenger` contract is non-upgradable, once a version of the contract is deployed it cannot be changed. This is with the intention of preventing any changes to the deployed contract that could potentially introduce bugs or vulnerabilities.
+
+However, there should still be new versions of the `TeleporterMessenger` contract that provide new features, and `TeleporterRegistry` supports the integration with these new versions.
```

Comment:
```suggestion
However, there could still be new versions of `TeleporterMessenger` deployed in the future. `TeleporterRegistry` is designed to enable applications using a `TeleporterMessenger` instance to integrate newer versions without minimal to no required interaction.
```

I don't really like my wording, but don't think we should specify "new versions that provide new features".

Reply:
updated